src/generators/dom/visitors/YieldTag.js

```diff
@@ -1,9 +1,11 @@
 export default function visitYieldTag ( generator, block, state ) {
-	block.builders.mount.addLine(
-		`${block.component}._yield && ${block.component}._yield.mount( ${state.parentNode || block.target}, null );`
+	const parentNode = state.parentNode || block.target;
+
+	( state.parentNode ? block.builders.create : block.builders.mount ).addLine(
+		`if ( ${block.component}._yield ) ${block.component}._yield.mount( ${parentNode}, null );`
```

Comment:
Is there a reason for switching this (and the destroy block below) to a regular `if` from a short circuited `&&`? Just a style thing? It seems we're relying on short circuiting in generated code for conciseness elsewhere.